CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/ReplicatorConfigurationRequestHandler.java

```diff
@@ -327,4 +362,177 @@ public boolean filtered(Document document, EnumSet<DocumentFlag> flags) {
     }
 }
 
+class LocalWinsCustomConflictResolver implements ConflictResolver {
+    private static final String TAG = "CCRREPLCONFIGHANDLER";
+    @Override
+    public Document resolve(Conflict conflict) {
+        Document localDoc = conflict.getLocalDocument();
+        Document remoteDoc = conflict.getRemoteDocument();
+        String localDocId = localDoc.getId();
+        String remoteDocId = remoteDoc.getId();
+        String docId = conflict.getDocumentId();
+        if (remoteDocId != localDocId) {
+            Log.e(TAG, "Remote docId and local docId are different");
+        }
+        if (remoteDocId != docId) {
+            Log.e(TAG, "Remote docId doesn't match with conflict docId");
+        }
+        if (docId != localDocId) {
+            Log.e(TAG, "Local docId doesn't match with conflict docId");
+        }
```

Comment:
2 questions: (1) in any case of doc id is not equal, do you need to notify the python test case in stead of just write log? (2) how do you know the conflict object contains the doc ids you expected?

Reply:
>> in any case of doc id is not equal, do you need to notify the python test case in stead of just write log?

We can't do any thing if the doc ids don't match. CBL has logic to autocorrect doc id, even if we enforced wrong doc id on CCR.

>> how do you know the conflict object contains the doc ids you expected

There is no way to know in advance that a particular doc is going to get conflicted, so we can't check doc id in app code. However, we are doing the validation in python code to check after raising the conflict, our CCR also work as expected.

Reply:
Added